packages/wallet/src/peer-connect/index.ts

```diff
@@ -0,0 +1,44 @@
+import { WalletAPI } from "../api.js";
+import * as Browser from "../browser/index.js";
+
+type WalletHandler = (walletId: string, wallet: WalletAPI) => void;
+export const mkPeerConnectAdapter = () => {
+  let wallet: WalletAPI | undefined;
+  let newWalletHandler: WalletHandler = () => {};
+  let deleteWalletHandler: WalletHandler = () => {};
+  return {
+    adaptApiEject(walletName: string, walletId: string) {
+      if (wallet) {
+        deleteWalletHandler(walletId, wallet);
+      }
+      wallet = undefined;
+    },
+    async adaptApiInject(walletName: string, walletId: string) {
+      const di = {
+        extension: await window.cardano[walletName.toLowerCase()].enable(),
+      };
+      const peerWallet = {
+        waitConfirmation: Browser.waitConfirmation(di),
+        signTx: Browser.signTx(di),
+        getChangeAddress: Browser.getChangeAddress(di),
+        getUsedAddresses: Browser.getUsedAddresses(di),
+        getCollaterals: Browser.getCollaterals(di),
+        getUTxOs: Browser.getUTxOs(di),
+        isMainnet: Browser.isMainnet(di),
+        getTokens: Browser.getTokens(di),
+        getLovelaces: Browser.getLovelaces(di),
+      };
+      wallet = peerWallet;
```

Comment:
why not using this function instead ? 
```
/**
 * Returns an instance of the browser wallet API for the specified wallet.
 * @param walletName - The name of the wallet to get an instance of.
 * @returns An instance of the BrowserWalletAPI class.
 */
export async function mkBrowserWallet(
  walletName: SupportedWalletName
): Promise<WalletAPI> {
  if (
    getInstalledWalletExtensions()
      .map((extension) => extension.name.toLowerCase())
      .includes(walletName.toLowerCase())
  ) {
    const extension = await window.cardano[walletName.toLowerCase()].enable();
    const di = { extension };
    return {
      waitConfirmation: waitConfirmation(di),
      signTx: signTx(di),
      getChangeAddress: getChangeAddress(di),
      getUsedAddresses: getUsedAddresses(di),
      getCollaterals: getCollaterals(di),
      getUTxOs: getUTxOs(di),
      isMainnet: isMainnet(di),
      getTokens: getTokens(di),
      getLovelaces: getLovelaces(di),
    };
  } else {
    throw new Error(`Wallet ${walletName} is not available in the browser`);
  }
}
```

Reply:
So, after you do the handshake, cardano peer connect offers a CIP-30 compatible API and I needed an adaptor from CIP-30 to Marlowe's WalletAPI.

This is not a browser wallet, so using that function would be confusing. I also did not wanted to change `getInstalledWalletExtensions` for the same reason.

Note that the actor in charge of creating the wallet is inverted from normal uses. We get a callback when somebody connects rather than the dapp being able to create a wallet connection at will.

Instead of exposing the inner functions we could create an `adaptCIP30Wallet` and use that instead.

If we want to add support for the walletConnect standard through adalib we can also [use the same adaptor](https://github.com/dcSpark/adalib/blob/development/docs/docs/Actions/adalib.md#enabling-a-connection)